### PR TITLE
Change logic for auto-fading connections

### PR DESCRIPTION
**Pull request by A-CGray**
### Previous behaviour

Currently, if the user specifies the `connected` option for fading connections then a connection will be faded if either the source or target component is faded. However, in the case where you are creating multiple copies of an XDSM with one component highlighted at a time, the result is that all connections are always faded because either the source or target is always faded (see the figure below)

image

### New behaviour 

This PR changes the default logic of the `connected` option so that a connection is faded only if the source **and** target are faded. This means that if a component is not faded, all of it's input and output connections are guaranteed to also be un-faded.

image

This is what I need for the applications I typically use fading for, but I'm sure there are plenty of other examples where the old behaviour would be preferable, so I'm open to feedback on whether this is a good change to make or not.

## Expected time until merged

1-2 weeks

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Comment by lamkina** [comment]
In the second figure, do you want the "hanging" incoming and outgoing connections to be faded as well?  As an example, in the second figure the incoming connection to the `Lift/Drag` off-diagonal block would be faded, but the connection between the `Lift/Drag` and `Aircraft Performance` target would be un-faded.

I think that would be a bit cleaner if it's not a total headache to implement.

**Comment by A-CGray** [comment]
Yeah that's a good call, I added attributes to the connection to track whether the source and target systems are faded and the connection edges are now faded based on that rather than whether the connection is faded. New behaviour looks like this

image

**Review by lamkina — APPROVED** [review]
Good stuff.  I think that's a clean solution.

**Review by nwu63 — APPROVED** [review]
I did not review the code, but the behaviour as described sounds good to me. Our typical workflow involves making a sequence of XDSM diagrams, highlighting each component one at a time, so this would be good. In the past I think I have used some hacky solution instead of the `connected` option to do this, but this automates a lot of that.

**Comment by nwu63** [comment]
I suggest bumping the minor version and making a release after this is merged.

**Review by nwu63 — DISMISSED** [review]

File: pyxdsm/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "2.2.2"
```
Comment (nwu63) [inline pyxdsm/__init__.py:1]: Wait you want `2.3.0` right?

**Review by nwu63 — APPROVED** [review]

**Pull request merged**